### PR TITLE
Add an annotation remover for serialization chunks

### PR DESCRIPTION
**Pull request by dslmeinte**

### CODE REVIEW

File: packages/utilities/src/serialization/annotation-remover.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ * @param languages - An array of {@link Language languages} to determine what nodes are instances of annotations
+ */
+export const withoutAnnotations = (serializationChunk: SerializationChunk, languages: Language[]) => {
+    const symbolTable = new MemoisingSymbolTable(languages)
+    const {serializationFormatVersion, languages: dependentLanguages, nodes} = serializationChunk
+    return {
+        serializationFormatVersion,
+        languages: dependentLanguages,
+        nodes: nodes
+            .filter((node) => !(symbolTable.entityMatching(node.classifier) instanceof Annotation)) // removes instances of annotations
```
Comment (joswarmer) [inline packages/utilities/src/serialization/annotation-remover.ts:17]: If the classifier cannot be found the node is not recognised as an Annotation node and the node will be kept in the resulting chunk. This chunk then is incorrect as such an annotation node will be in there without a parent.

An algorithm that does not have this problem would be to remove all annotation id's from each node's annotation array, keep the removed id's and them remove all the nodes with these id's.
The result is now guaranteed to be an (at least as) valid LionWeb chunk. 

And you don't even need the 'languages' parameter.

Comment (dslmeinte) [reply packages/utilities/src/serialization/annotation-remover.ts:17]: OK, will do that.

File: packages/utilities/src/serialization/annotation-remover.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {Annotation, Language, MemoisingSymbolTable, SerializationChunk} from "@lionweb/core"
+
+/**
+ * Removes all annotation-related instance data from the given {@link SerializationChunk}, i.e.:
+ *  * all nodes that are instances of an {@link Annotation},
+ *  * all annotation on nodes.
+ * @param serializationChunk - The {@link SerializationChunk}
+ * @param languages - An array of {@link Language languages} to determine what nodes are instances of annotations
```
[the remark this answers is not in the document]
Comment (dslmeinte) [reply packages/utilities/src/serialization/annotation-remover.ts:8]: See below.

File: packages/test/src/annotation-remover.test.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+    it("removes annotation instances and references to it", () => {
+        const annoLanguage = new Language("annoLanguage", "0", "annoLanguage", "annoLanguage")
+        const aConcept = new Concept(annoLanguage, "aConcept", "aConcept", "aConcept", false)
+        const anAnnotation = new Annotation(annoLanguage, "anAnno", "anAnno", "anAnno").annotating(aConcept)
+        annoLanguage.havingEntities(aConcept, anAnnotation)
+
+        const serializationChunk: SerializationChunk = {
+            serializationFormatVersion: currentSerializationFormatVersion,
+            languages: [],
```
[the remark this answers is not in the document]
Comment (dslmeinte) [reply packages/test/src/annotation-remover.test.ts:18]: Will fix that: only the LionCore M3 needs to be in.

File: packages/utilities/src/serialization/annotation-remover.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ */
+export const withoutAnnotations = (serializationChunk: SerializationChunk, languages: Language[]) => {
+    const symbolTable = new MemoisingSymbolTable(languages)
+    const {serializationFormatVersion, languages: dependentLanguages, nodes} = serializationChunk
```
[the remark this answers is not in the document]
Comment (dslmeinte) [reply packages/utilities/src/serialization/annotation-remover.ts:12]: See below: the `languageList` parameter will be removed anyway.

**Review by joswarmer — APPROVED** [review]

**Pull request merged**